src/pe/optional_header.rs

```diff
@@ -234,6 +235,40 @@ impl From<WindowsFields32> for WindowsFields {
 
 pub type WindowsFields = WindowsFields64;
 
+impl WindowsFields {
+    /// Validate the Windows-Specific Fields
+    ///
+    /// This method DOES NOT validate `check_sum` field since
+    /// the algorithm for computing the checksum is incorporated
+    /// into IMAGEHELP.DLL, which is os-dependent.
+    pub fn validate(&self) -> error::Result<()> {
+        let mut error_messages = vec![];
+        if self.image_base % 0x1000 != 0 {
+            error_messages.push("ImageBase must be a multiple of 64K.".to_owned());
+        }
+        if self.section_alignment < self.file_alignment {
+            error_messages.push(
+                "SectionAlignment must be greater than or equal to FileAlignment.".to_owned(),
+            );
+        }
+        if self.file_alignment < 512 || self.file_alignment > 0x1000 || self.file_alignment % 2 != 0
```

Comment:
This check validates e.g. 514 despite not being a power of two. In addition to that it should also be 0x1000**0** instead of 0x1000.

It should be `if self.file_alignment < 0x200 || self.file_alignment > 0x10000 || (self.file_alignment & (self.file_alignment - 1)) == 0` to validate all requirements.